api/app/Rules/HasEducationRequirement.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+use App\Models\PoolCandidate;
+
+class HasEducationRequirement implements Rule
+{
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string $attribute
+     * @param  mixed $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        // short-circuit check off feature flag
+        $flagBoolean = config('feature.application_revamp');
+        if (!$flagBoolean) {
+            return true;
+        }
+
+        $poolCandidate = PoolCandidate::findOrFail($value);
+        if (!$poolCandidate->education_requirement_option) {
+            return false;
+        }
+        $experiences = $poolCandidate->getEducationRequirementExperiencesAttribute();
+        return isset($experiences) && (count($experiences) >= 1);
+    }
+
+    /**
+     * Get the validation error message.
+     *
+     * @return string
+     */
+    public function message()
+    {
+        return 'EducationRequirementIncomplete';
```

Comment:
Can we make this part of the enum that handles pool candidate error codes?

https://github.com/GCTC-NTGC/gc-digital-talent/blob/9a93c576b3fa9d1f1a29e86f027d6531691b72a3/api/database/helpers/ApiEnums.php#L405-L427

Reply:
Enum stuff done here https://github.com/GCTC-NTGC/gc-digital-talent/pull/6458/commits/cdcb50ab9d2d1d3811e35600c275845e372418b2
Message added too, I realized the validation error messages are probably never going to be seen since it relies on the front-end validation failing but back-ups are back-ups https://github.com/GCTC-NTGC/gc-digital-talent/pull/6458/commits/ce4549f0dfa65b36b041e9488dad007b88b89d5e

As a thrown in thing, I noticed one other PHP test was failing, looked at it and it seemed like a simple thing. Sometimes the clock would roll to the next second while it was running, so I fixed things by just comparing to the nearest minute rather than second. https://github.com/GCTC-NTGC/gc-digital-talent/pull/6458/commits/f6aefb8c02bb2c2b7b76207ec3c0322c9bd1e5c7

Reply:
And got the translation in for it quickly with https://github.com/GCTC-NTGC/gc-digital-talent/pull/6458/commits/cfcd70254a331fe13442e602ca5d635b7037dac5